### PR TITLE
CIV-9939 Payment url

### PR DESCRIPTION
**Pull request by ElifDenizEsea**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-9939
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/routes/features/claim/payment/claimFeeBreakDownController.ts [hunk 1]
```diff
@@ -24,4 +26,19 @@ claimFeeBreakDownController.get(CLAIM_FEE_BREAKUP, (async (req: AppRequest, res:
   }
 })as RequestHandler);
 
+claimFeeBreakDownController.post(CLAIM_FEE_BREAKUP, async (req:   AppRequest, res: Response, next: NextFunction) => {
+  try {
+    const claimId = req.params.id;
+    const paymentRedirectInformation = await getFeePaymentRedirectInformation(claimId, FeeType.CLAIM, req);
+
+    const claim = await getCaseDataFromStore(claimId);
```
Comment (jeswanth-hmcts) [inline src/main/routes/features/claim/payment/claimFeeBreakDownController.ts:34]: Correct me  if I am wrong , but are we not  using here to get  case details with userid+caseid as key

**Review by jeswanth-hmcts — DISMISSED** [review]

File: src/main/common/form/models/helpWithFees/feeType.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
 export enum FeeType {
   HEARING = 'HEARING',
+  CLAIM = 'CLAIM',
```
Comment (neeta-hmcts) [inline src/main/common/form/models/helpWithFees/feeType.ts:3]: In Civil service the feeType used is 'CLAIMISSUED'.. we should use the same in CUI for consistency

File: src/main/common/form/models/helpWithFees/feeType.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
 export enum FeeType {
   HEARING = 'HEARING',
+  CLAIM = 'CLAIMISSUED',
```
Comment (paul-pearson) [inline src/main/common/form/models/helpWithFees/feeType.ts:3]: It would be best to use `CLAIMISSUED = 'CLAIMISSUED',`

**Review by paul-pearson — CHANGES_REQUESTED** [review]

File: src/main/routes/features/claim/payment/claimFeeBreakDownController.ts [hunk 4]
```diff
@@ -24,4 +27,20 @@ claimFeeBreakDownController.get(CLAIM_FEE_BREAKUP, (async (req: AppRequest, res:
   }
 })as RequestHandler);
 
+claimFeeBreakDownController.post(CLAIM_FEE_BREAKUP, async (req:   AppRequest, res: Response, next: NextFunction) => {
+  try {
+    const claimId = req.params.id;
+    const paymentRedirectInformation = await getFeePaymentRedirectInformation(claimId, FeeType.CLAIM, req);
```
Comment (neeta-hmcts) [inline src/main/routes/features/claim/payment/claimFeeBreakDownController.ts:33]: type is incorrect

File: src/main/routes/features/claim/payment/claimFeeBreakDownController.ts (hunk 1 above)
Comment (ElifDenizEsea) [reply src/main/routes/features/claim/payment/claimFeeBreakDownController.ts]: done

File: src/main/common/form/models/helpWithFees/feeType.ts (hunk 3 above)
Comment (ElifDenizEsea) [reply src/main/common/form/models/helpWithFees/feeType.ts:3]: done

File: src/main/routes/features/claim/payment/claimFeeBreakDownController.ts (hunk 4 above)
Comment (ElifDenizEsea) [reply src/main/routes/features/claim/payment/claimFeeBreakDownController.ts:33]: done

File: src/main/routes/features/claim/payment/claimFeeBreakDownController.ts [hunk 5]
```diff
@@ -24,4 +26,18 @@ claimFeeBreakDownController.get(CLAIM_FEE_BREAKUP, (async (req: AppRequest, res:
   }
 })as RequestHandler);
 
+claimFeeBreakDownController.post(CLAIM_FEE_BREAKUP, async (req:   AppRequest, res: Response, next: NextFunction) => {
+  try {
+    const claimId = req.params.id;
```
[the remark this answers is not in the document]
Comment (ElifDenizEsea) [reply src/main/routes/features/claim/payment/claimFeeBreakDownController.ts:31]: it is used

File: src/main/routes/features/claim/payment/claimFeeBreakDownController.ts [hunk 6]
```diff
@@ -24,4 +26,18 @@ claimFeeBreakDownController.get(CLAIM_FEE_BREAKUP, (async (req: AppRequest, res:
...
+    const paymentRedirectInformation = await getFeePaymentRedirectInformation(claimId, FeeType.CLAIMISSUED , req);
+    const claim =  await getCaseDataFromStore(generateRedisKey(req));
+    claim.claimDetails.claimFeePayment=paymentRedirectInformation;
+    await saveDraftClaim(claim.id, claim, true);
+    res.redirect(paymentRedirectInformation?.nextUrl);
+
```
[the remark this answers is not in the document]
Comment (ElifDenizEsea) [reply src/main/routes/features/claim/payment/claimFeeBreakDownController.ts]: done

**Review by neeta-hmcts — APPROVED** [review]

**Review by neeta-hmcts — CHANGES_REQUESTED** [review]
As discussed this will need backend changes too,

**Review by neeta-hmcts — APPROVED** [review]

**Pull request closed without merging**